tests/test_algorithm/test_cg.py

```diff
@@ -93,6 +111,23 @@ def test_nodes_attain_same_value(self):
         self.assertFalse(nodes_attain_same_value(figure_9b.graph, event, X @ -d, D))
         self.assertFalse(nodes_attain_same_value(figure_9b.graph, event, D, X @ -d))
         self.assertFalse(nodes_attain_same_value(figure_9b.graph, event, D @ -d, D))
+        # This should get us onto the return False on line 56
+        self.assertFalse(nodes_attain_same_value(figure_9b.graph, event, Y @ +X, X))
+        # This one should trigger the False on line 59
+        self.assertFalse(nodes_attain_same_value(figure_9b.graph, {Y @ +X: +Y, Y: -Y}, Y @ +X, Y))
+        # This one will trigger the True on line 61, but I'm not sure if the event really makes sense
```

Comment:
If there are branches that you don't think make sense / it's impossible to construct a meaningful scenario to run them, we can switch return values for `raise RuntimeError` and explain why we don't think this branch makes sense